src/app/login/login-page/login-page.component.ts

```diff
@@ -1,16 +1,29 @@
 import { Component, OnInit } from '@angular/core';
-import { OAuthService } from 'angular-oauth2-oidc';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { AppAuthService } from '../../services/app-auth.service';
 
 @Component({
   selector: 'app-login-page',
-  templateUrl: './login-page.component.html'
+  templateUrl: './login-page.component.html',
 })
-export class LoginPageComponent {
-  constructor(
-    private readonly oAuthService: OAuthService,
-  ) { }
+export class LoginPageComponent implements OnInit {
+  loginForm!: FormGroup;
 
-  login(): void {
-    this.oAuthService.initLoginFlow();
+  constructor(private authService: AppAuthService, private formBuilder: FormBuilder) {}
+
+  ngOnInit() {
+    this.loginForm = this.formBuilder.group({
```

Comment:
this can be defined in the constructor or where the property is defined because you don't need to wait for the component to setup these values. By doing this, you will no longer need **ngOnInit**

Reply:
updated this